packages/rum-core/test/error-logging/error-logging.spec.js

```diff
@@ -285,4 +286,61 @@ describe('ErrorLogging', function() {
       expect(apmServer.errorQueue.items.length).toBe(4)
     }
   })
+
+  it('should capture unhandled rejection events', done => {
+    configService.set('flushInterval', 1)
+    errorLogging.registerListeners()
+
+    spyOn(apmServer, 'sendErrors').and.callFake(errors => {
+      expect(errors[0].exception.message).toMatch(reason.message)
+      done()
+    })
+    /**
+     * simulate window.PromiseRejectionEvent event since its not supported by
+     * all browsers
+     */
+    const reason = new Error(testErrorMessage)
+    const event = new CustomEvent('unhandledrejection')
+    event.reason = reason
+    window.dispatchEvent(event)
+  })
+
+  it('should handle multiple reasons for promise rejections', () => {
+    const getErrors = () => apmServer.errorQueue.items
+
+    errorLogging.logPromiseEvent({})
+    expect(getErrors().length).toBe(1)
+    expect(getErrors()[0].exception.message).toMatch(/no reason specified/)
+
+    const error = new Error(testErrorMessage)
+    errorLogging.logPromiseEvent({
+      reason: error
+    })
+    expect(getErrors()[1].exception.message).toMatch(error.message)
+
+    errorLogging.logPromiseEvent({
+      reason: testErrorMessage
+    })
+    expect(getErrors()[2].exception.message).toMatch(testErrorMessage)
+
+    const errorObj = {
+      message: testErrorMessage,
+      stack: 'ReferenceError: At example.js:23'
+    }
+    errorLogging.logPromiseEvent({
+      reason: errorObj
+    })
+    expect(getErrors()[3].exception.message).toMatch(testErrorMessage)
+    expect(getErrors()[3].exception.stacktrace.length).toBeGreaterThan(0)
+
+    const errorLikeObj = {
+      message: testErrorMessage,
+      foo: 'bar'
+    }
+    errorLogging.logPromiseEvent({
+      reason: errorLikeObj
+    })
+    expect(getErrors()[4].exception.message).toMatch(testErrorMessage)
+    expect(getErrors()[4].exception.stacktrace.length).toBe(0)
```

Comment:
We should expand this test to include `reason` that are arbitrary objects or even other types! Because `reason` can be created by user code and it means it can have any value!